atk/src/auto/object.rs

```diff
@@ -814,19 +821,27 @@ impl<O: IsA<Object>> AtkObjectExt for O {
         }
         unsafe {
             let f: Box_<F> = Box_::new(f);
-            connect_raw(
+            let detailed_signal_name =
+                detail.map(|name| format!("active-descendant-changed::{}\0", name));
+            let signal_name: &[u8] = detailed_signal_name
+                .as_ref()
+                .map_or(&b"active-descendant-changed\0"[..], |n| n.as_bytes());
+            let handler_id = connect_raw(
                 self.as_ptr() as *mut _,
-                b"active-descendant-changed\0".as_ptr() as *const _,
+                signal_name.as_ptr() as *const _,
                 Some(transmute::<_, unsafe extern "C" fn()>(
                     active_descendant_changed_trampoline::<Self, F> as *const (),
                 )),
                 Box_::into_raw(f),
-            )
+            );
+            drop(detailed_signal_name);
+            handler_id
```

Comment:
As mentioned in the gir review you really don't need to drop `detailed_signal_name` here; this is the end of the function where it goes out of scope and is cleaned up by definition :)